src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -155,6 +159,11 @@ private async Task MonitorRuleChangesAsync(CancellationToken token)
 
                 _logger.CollectionRuleConfigurationChanged();
 
+                if (_portOptions.ConnectionMode != DiagnosticPortConnectionMode.Listen || string.IsNullOrEmpty(_portOptions.EndpointName))
```

Comment:
The [IValidateOptions<DiagnosticPortOptions> implementation](https://github.com/dotnet/dotnet-monitor/blob/main/src/Tools/dotnet-monitor/DiagnosticPortValidateOptions.cs) already validates that the endpoint was specified if the connection mode is `Listen`. Thus, you will not be able to get an instance of `DiagnosticPortOptions` through the options system if this condition is not satisfied. You do not need to check this condition here and can just check the connection mode. I would suggest using [DiagnosticPortOptionsExtensions.GetConnectionMode](https://github.com/dotnet/dotnet-monitor/blob/main/src/Microsoft.Diagnostics.Monitoring.Options/DiagnosticPortOptionsExtensions.cs) to get its understood value.